proposals/NNNN-hlsl-matrix-type.md

```diff
@@ -0,0 +1,136 @@
+<!-- {% raw %} -->
+
+# HLSL Constant Matrices
+
+* Proposal: [NNNN](NNNN-hlsl-matrix-type.md)
+* Author(s): [Greg Roth](https://github.com/pow2clk)
+* Status: **Design In Progress**
+
+* Issues:
+  [#88060](https://github.com/llvm/llvm-project/issues/88060)
+
+## Introduction
+
+LLVM matrix support differs in some key ways from the features and limiations
+of HLSL matrices. This proposes inheriting from the existing matrix type that
+will be enabled in HLSL by default and will impose these limits and allow the
+additional capabilities.
+
+## Motivation
+
+Whether transforming geometric coordinates or adapting weights for machine
+learning, matrix support is crucial to the existing and future utility of HLSL.
+Fortunately, Clang and LLVM support an extension that provides
+the fundamentals needed for HLSL matrix support.
+However, it is too permissive in the allowable dimensions
+and lacking in various other ways that the matrices can be used.
+
+HLSL matrices in DXC have the following features that Clang matrices lack:
```

Comment:
Can you provide some HLSL examples along with some of these descriptions to help better illustrate what is being presented here.  You have some syntax here already which is nice.